### ISSUE TITLE
Megadropdown focus order

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#4271]
The close button should be the last element in focus order when the megadropdown opens. The order should be:

1. Megadropdown trigger is pressed
2. Tab should go to the first focusable link in the dropdown which is either the title or a link in the list
3. After reaching the end of the list, focus should jump to the close button
4. When pressing tab, focus should jump to the next megadropdown trigger or main navigation link if there are any, otherwise it should go to the next focusable element on the page

<img width="1128" alt="Image" src="https://github.com/user-attachments/assets/0ef5a744-74ab-4280-b68b-004fca2f7212" />
